examples/MagicWeatherCompose/app/src/main/java/com/revenuecat/sample/paywall/PaywallView.kt

```diff
@@ -0,0 +1,102 @@
+package com.revenuecat.sample.paywall
+
+import android.app.Activity
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.material3.Button
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.sample.utils.buttonText
+import com.revenuecat.sample.utils.findActivity
+
+@Suppress("LongParameterList")
+@Composable
+fun PaywallView(
+    offering: Offering,
+    modifier: Modifier = Modifier,
+    onPurchaseStarted: ((Package) -> Unit)? = null,
+    onPurchaseCompleted: ((CustomerInfo) -> Unit)? = null,
+    onPurchaseCancelled: (() -> Unit)? = null,
+    onPurchaseErrored: ((PurchasesError) -> Unit)? = null,
+    marketingContent: (@Composable () -> Unit)? = null,
```

Comment:
Not a blocker, but do you think there's a way we can remove the number of parameters of this function?

Reply:
Well, thinking about it, we definitely should add a listener of some sort, instead of adding all those callbacks. Might do that in a different PR. That would join 4 parameters into 1. As for the others, I think they need to be there, but we can think of ways to refactor.